modules/ml/src/main/java/org/apache/ignite/ml/math/VectorUtils.java

```diff
@@ -154,4 +158,38 @@ public static Vector copyPart(Vector v, int off, int len) {
 
         return res;
     }
+
+    /**
+     * Creates dense local on heap vector based on array of doubles.
+     *
+     * @param values Values.
+     */
+    public static Vector of(double ... values) {
+        A.notNull(values, "values");
+
+        return new DenseLocalOnHeapVector(values);
+    }
+
+    /**
+     * Creates vector based on array of Doubles. If array contains null-elements then
+     * method returns sparse local on head vector. In other case method returns
+     * dense local on heap vector.
+     *
+     * @param values Values.
+     */
+    public static Vector of(Double[] values) {
+        A.notNull(values, "values");
+
+        Vector answer = null;
+        if (Arrays.stream(values).anyMatch(Objects::isNull))
+            answer = new SparseLocalVector(values.length, StorageConstants.RANDOM_ACCESS_MODE);
```

Comment:
maybe doing this more clearly with special flag (isSparse)? I don't catch the logic here